backend/src/utils/db_utils.py

```
@@ -1229,9 +1229,95 @@ def delete_category_from_db(category_id: uuid.UUID, user_id: str) -> bool:
+                    update_transaction(transaction)
+                    cleaned_count += 1
+            
+            # If we got fewer transactions than requested, we're done
+            if not last_evaluated_key:
+                break
+        
+        return cleaned_count
+        
+    except Exception as e:
+        logger.error(f"DB: Error cleaning up transaction references for category {str(category_id)}: {str(e)}", exc_info=True)
+        raise
```

Pylint (3.3.7)

[refactor] Too many nested blocks (6/5)

(R1702)